packages/amplify_datastore/ios/Classes/SwiftAmplifyDataStorePlugin.swift

```diff
@@ -131,6 +133,57 @@ public class SwiftAmplifyDataStorePlugin: NSObject, FlutterPlugin {
             return
         }
     }
+
+    func onDelete(args: [String: Any], flutterResult: @escaping FlutterResult) {
+        var modelName: String
+        var id: String
+        var rawModel: Dictionary<String, Any>
+        var modelData: SerializedModel
+        
+        do {
+            modelName = args["modelName"] as! String
```

Comment:
It is always best to avoid force casting, if for some reason `args` do not contains the key, this line will crash the app. So to be on the safe side you can guard this as line and return an error if the required information is not present.
```swift
guard let modelName = args["modelName"] as? String, 
           let rawModel = args["model"] as? Dictionary<String, Any>,
           let id = rawModel["id"] as? String else {
      
          flutterResult(error)
          return
}

let modelData = SerializedModel(id: id, map: getJsonValue(rawModel))

do {
...
} catch {

}
           
```

Reply:
I ended up using individual guard statements.